clang/lib/CIR/Lowering/DirectToLLVM/LowerToLLVM.cpp

```diff
@@ -4457,6 +4457,30 @@ mlir::LogicalResult CIRToLLVMLinkerOptionsOpLowering::matchAndRewrite(
   return mlir::success();
 }
 
+mlir::LogicalResult CIRToLLVMBlockAddressOpLowering::matchAndRewrite(
+    cir::BlockAddressOp op, OpAdaptor adaptor,
+    mlir::ConversionPatternRewriter &rewriter) const {
+  cir::LabelOp labelOp;
+  mlir::MLIRContext *ctx = rewriter.getContext();
+
+  op.findLabel(labelOp);
```

Comment:
The LLVM lowering can benefit from the same style of bookkeeping mentioned for CIRGen, see `stringGlobalsMap` for inspiration on how to access module state.